uniffi_meta/src/types.rs

```diff
@@ -141,6 +141,19 @@ impl Type {
         };
         Box::new(std::iter::once(self).chain(nested_types))
     }
+
+    pub fn needs_unchecked_upcast_in_return(&self) -> bool {
```

Comment:
I don't quite follow this - why does only a `Map<Object, Object>` need this?

Reply:
It's actually a Map where either key or value or both types are objects. In Kotlin, `Map<Int, FooInterface>` and `Map<Int, Foo>` are different types, so when we read those from Rust through `lift` we get a `Map<Int, Foo>` but the return type of the function is actually `Map<Int, FooInterface>` and it can't be casted implicitly.

Reply:
Surprisingly, I didn't find this same issue with `List`. I'll double check, just in case.